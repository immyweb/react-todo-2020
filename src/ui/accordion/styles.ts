import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { mediaQueries } from '../../styles/breakpoints';
import { transparentize } from 'polished';

import { H4, H5, A } from '../../ui/typography/typography';

import { ITheme } from '../../types/theme';

// Accordion

export const AccordionContainer = glamorous.section({
  label: 'accordion-container',
  position: 'relative',
  marginBottom: 25,
  background: colours.$white,
  ':after': {
    clear: 'both',
    content: '""',
    display: 'table',
  },
});

export const Border = glamorous.div<{ colour: string }>(
  {
    label: 'border',
    left: 0,
    content: '""',
    position: 'absolute',
    top: 0,
    height: '100%',
    width: 5,
    zIndex: 10,
  },
  ({ colour }) => ({
    background: colour,
  }),
);

// Accordion Item

export const Item = glamorous.div<{
  open: boolean;
  index: number;
  theme: ITheme;
  border?: boolean;
}>(
  {
    label: 'accordion-item',
    position: 'relative',
    overflowY: 'hidden',
    transition: 'border-color 250ms ease',
    ':before': {
      content: '""',
      position: 'absolute',
      height: 0,
      width: 0,
      top: 22,
      left: 5,
      border: '4px solid transparent',
    },
    ':last-child': {
      borderBottom: 0,
    },
  },
  ({ open, index, theme, border }) => ({
    borderLeft: border ? `5px solid ${theme.secondary}` : 'none',
    borderBottom: open
      ? `2px solid ${transparentize(0.8, theme.primary)}`
      : `2px solid ${colours.$white}`,
    label: `${index}--accordion-item`,
  }),
);

// Accordion Item Head

export const Header = glamorous.header<{ open: boolean; colour: string }>(
  {
    label: 'accordion-header',
    display: 'flex',
    flexDirection: 'row',
    alignItems: 'center',
    padding: '15px 20px',
    minHeight: 80,
    color: colours.$offBlack,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ open, colour }) => ({
    background: open ? colours.$white : transparentize(0.8, colour),
  }),
);

export const BadgeHolder = glamorous.div({
  label: 'badge-holder',
  display: 'flex',
  borderRadius: '50%',
  overflow: 'hidden',
  position: 'relative',
});

export const Badge = glamorous.img({
  label: 'badge',
  width: 50,
  height: 50,
});

export const TextHolder = glamorous.div<{ hasBadge?: boolean }>(
  {
    label: 'title-text',
    flexDirection: 'column',
    position: 'relative',
    paddingRight: 15,
  },
  ({ hasBadge }) => ({
    paddingLeft: hasBadge ? 15 : 0,
    flexBasis: hasBadge ? `calc(100% - 65px)` : '100%',
  }),
);

export const SubTitle = glamorous(H5)({
  label: 'accordion-date',
  marginBottom: 0,
});

export const Title = glamorous(H4)<{ index: number }>(
  {
    label: 'heading',
    marginBottom: 0,
    position: 'relative',
  },
  ({ index }) => ({
    label: `${index}--accordion-h4`,
  }),
);

// Accordion Item Body

export const defaultStyle = {
  transition: `opacity ${150}ms ease-out`,
  opacity: 0,
};

export const transitionStyles = {
  entering: { opacity: 0 },
  entered: { opacity: 1 },
  exiting: { opacity: 1 },
  exited: { opacity: 0 },
};

export const Panel = glamorous.article<{ open: boolean }>(
  {
    label: 'accordion-panel',
    padding: '0 20px',
    overflowY: 'hidden',
    [mediaQueries.tabletPortrait]: {
      borderBottom: 0,
    },
  },
  ({ open }) => ({
    display: open ? 'block' : 'none',
  }),
);

export const Copy = glamorous.div<{ colour: string }>(
  {
    label: 'accordion-copy',
    margin: '0 0 15px',
    '& a': {
      textDecoration: 'none',
    },
  },
  ({ colour }) => ({
    '& a': {
      color: colour,
    },
  }),
);

export const Link = glamorous(A)<{ colour: string }>(
  {
    label: 'accordion-link',
    display: 'inline-block',
  },
  ({ colour }) => ({
    color: colour,
  }),
);
