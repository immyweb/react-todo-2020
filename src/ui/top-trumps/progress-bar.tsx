import * as React from 'react';

import { ICard } from './types';
import { ITheme } from '../../types/theme';
import {
  TTProgressBarContainer,
  TTProgressBarInnerContainer,
  TTProgressBarCircle,
  TTProgressBarArrowContainer,
  TTProgressBarArrow,
} from './styles';

interface IProps {
  className: string;
  cards: ICard[];
  theme: ITheme;
  currentCardIndex: number;
  onClickRight: () => void;
  onClickLeft: () => void;
}

const onClickConditional = (disabled: boolean, func: IProps['onClickLeft']) => {
  if (!disabled) {
    func();
  }
};

export const ProgressBar: React.SFC<IProps> = ({
  cards,
  theme,
  onClickLeft,
  onClickRight,
  currentCardIndex,
}) => {
  const items = cards.map((card, i) => {
    const { id } = card;

    return (
      <TTProgressBarCircle
        viewed={i === currentCardIndex}
        inactiveColour={theme.primary}
        className={'tt-progress'}
        key={id}
      />
    );
  });

  const firstArrowDisabled = currentCardIndex === 0;

  return (
    <TTProgressBarContainer
      className="tt-progress_bar-container"
      borderColour={theme.primary}
      bkgnd={theme.secondary}
    >
      <TTProgressBarArrowContainer
        right={false}
        onClick={() => onClickConditional(firstArrowDisabled, onClickLeft)}
      >
        <TTProgressBarArrow disabled={firstArrowDisabled} right={false} />
      </TTProgressBarArrowContainer>
      <TTProgressBarInnerContainer>{items}</TTProgressBarInnerContainer>
      <TTProgressBarArrowContainer right={true} onClick={onClickRight}>
        <TTProgressBarArrow disabled={false} right={true} />
      </TTProgressBarArrowContainer>
    </TTProgressBarContainer>
  );
};
