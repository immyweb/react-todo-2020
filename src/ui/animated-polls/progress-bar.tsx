import * as React from 'react';

import { APProgressBar, APProgressBarItem } from './styles';

import { ICard } from './types';

interface IProps {
  options: ICard[];
  currentCardIndex: number;
}

export const ProgressBar: React.SFC<IProps> = ({ options, currentCardIndex }) => {
  const items = options.map((option, i) => {
    const { id } = option;

    return <APProgressBarItem viewed={i === currentCardIndex} key={id} />;
  });
  return <APProgressBar>{items}</APProgressBar>;
};
