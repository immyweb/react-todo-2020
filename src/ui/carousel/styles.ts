import glamorous from 'glamorous';

import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { BtnNoStyle } from '../../styles/globals';
import { Direction } from './types';

export const CarouselContainer = glamorous.section({
  label: 'carousel-container',
  paddingBottom: 33,
  marginBottom: 25,
  position: 'relative',
  width: '100%',
});

export const Slide = glamorous.div({
  label: 'carousel-slide',
  cursor: 'pointer',
  padding: '0 10px',
  ':focus': {
    outline: 'none',
  },
});

export const SlideLink = glamorous.a({
  label: 'carousel-link',
  display: 'inline-block',
  width: '100%',
  height: '100%',
  objectFit: 'contain',
  textDecoration: 'none',
});

export const SlideImg = glamorous.img<{ borderColour: string }>(
  {
    label: 'carousel-image',
    width: '100%',
    height: '100%',
    objectFit: 'contain',
  },
  ({ borderColour }) => ({
    borderBottom: `3px solid ${borderColour}`,
  }),
);

export const SlideLabel = glamorous.p<{ txtColour: string }>(
  {
    label: 'carousel-label',
    textAlign: 'center',
    fontFamily: fonts.$theSunHeavyNarrow,
    fontSize: 18,
    paddingTop: 8,
  },
  ({ txtColour }) => ({
    color: txtColour,
  }),
);

export const SlideNav = glamorous(BtnNoStyle)<{ direction: Direction }>(
  {
    label: 'carousel-nav',
    position: 'absolute',
    zIndex: 3,
    bottom: -47,
    width: 48,
    height: 48,
    ':hover': {
      cursor: 'pointer',
    },
  },
  ({ direction }) => ({
    left: direction === Direction.PREV ? 25 : 'auto',
    right: direction === Direction.NEXT ? 25 : 'auto',
  }),
);

// Pagination

export const PaginationContainer = glamorous.div({
  label: 'pagination-container',
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
  width: '100%',
  height: 20,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const Dot = glamorous.span<{ colour?: string; label?: string }>(
  {
    display: 'inline-block',
    borderRadius: '50%',
    margin: '0 5px',
  },
  ({ colour, label }) => ({
    label,
    background: colour ? colour : colours.$news,
  }),
);
