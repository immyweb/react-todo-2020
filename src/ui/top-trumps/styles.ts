import glamorous from 'glamorous';
import { colours } from '../../styles/colours';
import { fonts } from '../../styles/fonts';
import { H3, H2 } from '../typography/typography';
import { mediaQueries } from '../../styles/breakpoints';

export const TTContainer = glamorous.section({
  label: 'tt-container',
  width: '100%',
  paddingTop: '134.9%',
  minHeight: 400,
  minWidth: 280,
  backgroundColor: colours.$black,
  position: 'relative',
  [mediaQueries.mobileInner]: {
    paddingTop: '72.9%',
  },
});

export const TTIntroInner = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'tt-intro-inner',
    width: '100%',
    height: '100%',
    position: 'absolute',
    top: 0,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const TTDefaultInner = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'tt-default-inner',
    width: '100%',
    height: '100%',
    position: 'absolute',
    top: 0,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const TTVoteWrapper = glamorous.div<{}>({
  label: 'tt-vote-wrapper',
  display: 'flex',
});

export const TTWrapper = glamorous.section({
  label: 'tt-wrapper',
  margin: '0 5%',
  height: '100%',
  width: '90%',
  display: 'flex',
  flexDirection: 'column',
  alignItems: 'center',
  position: 'relative',
  [mediaQueries.mobileInner]: {
    margin: '0 5%',
  },
});

export const TTTopGradient = glamorous.div({
  label: 'ttt-top-gradient',
  width: '100%',
  height: 84,
  background: `linear-gradient(to bottom, ${colours.$offBlack}, transparent)`,
  position: 'absolute',
  top: 0,
  zIndex: 2,
});

export const TTBottomGradient = glamorous.div({
  label: 'ttt-top-gradient',
  width: '100%',
  height: 130,
  background: `linear-gradient(rgba(34, 37, 38, 0.9), rgba(34, 37, 38, 0.9))`,
  position: 'absolute',
  bottom: 0,
  zIndex: 2,
});

export const TTHeadline = glamorous(H3)({
  label: 'ttt-card-first-name',
  fontSize: 18,
  textTransform: 'uppercase',
  textAlign: 'center',
  marginTop: 0,
  fontFamily: fonts.$theSunHeavyCondensed,
  [mediaQueries.mobileInner]: {
    fontSize: 18,
  },
});

export const TTSubHeadline = glamorous(H2)({
  label: 'ttt-card-surname',
  fontSize: 48,
  textTransform: 'uppercase',
  textAlign: 'center',
  [mediaQueries.mobileInner]: {
    fontSize: 46,
  },
});

export const TTButton = glamorous.div<{
  backgroundColour: string;
  hoverBackgroundColour: string;
}>(
  {
    label: 'tt-intro-button',
    display: 'flex',
    margin: 'auto',
    alignItems: 'center',
    justifyContent: 'center',
    fontFamily: fonts.$theSunMedium,
    textAlign: 'center',
    height: 44,
    color: colours.$white,
    [mediaQueries.mobileInner]: {
      width: 250,
    },
  },
  ({ backgroundColour, hoverBackgroundColour }) => ({
    backgroundColor: backgroundColour,
    ':hover': {
      backgroundColor: hoverBackgroundColour,
    },
  }),
);

export const TTIntroButtonHolder = glamorous.div({
  label: 'tt-intro-button-holder',
  position: 'absolute',
  width: '100%',
  bottom: 15,
  cursor: 'pointer',
});

export const TTGame = glamorous.section({
  label: 'tt-game',
});

export const TTCard = glamorous.div<{
  imageMobile: string;
  imageDesktop: string;
}>(
  {
    label: 'tt-card',
    width: '100%',
    height: '91.5%',
    position: 'absolute',
    top: 0,
    backgroundSize: 'cover',
    backgroundPosition: 'top center',
    backgroundRepeat: 'no-repeat',
    [mediaQueries.mobileInner]: {},
  },
  ({ imageMobile, imageDesktop }) => ({
    backgroundImage: `url(${imageMobile})`,
    [mediaQueries.mobileInner]: {
      backgroundImage: `url(${imageDesktop})`,
    },
  }),
);

export const TTProgressBarCircle = glamorous.div<{
  viewed: boolean;
  inactiveColour: string;
}>(
  {
    label: 'tt-progress_bar-circle',
    display: 'inline-block',
    borderRadius: '50%',
  },
  ({ viewed, inactiveColour }) => ({
    backgroundColor: viewed ? colours.$white : inactiveColour,
    width: viewed ? 11 : 7,
    height: viewed ? 11 : 7,
  }),
);

export const TTProgressBarContainer = glamorous.div<{
  bkgnd: string;
  borderColour: string;
}>(
  {
    label: 'tt-progress_bar-container',
    display: 'flex',
    position: 'absolute',
    bottom: 0,
    justifyContent: 'space-between',
    height: '8.5%',
    width: '100%',
    textAlign: 'center',
  },
  ({ bkgnd, borderColour }) => ({
    background: bkgnd,
    borderTop: `1px solid ${borderColour}`,
  }),
);

export const TTProgressBarInnerContainer = glamorous.div({
  label: 'tt-progress_bar-inner-container',
  width: 130,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'space-evenly',
});

export const TTProgressBarArrowContainer = glamorous.div<{
  right: boolean;
}>(
  {
    label: 'tt-progress_bar-arrow-container',
    padding: 15,
    width: 45,
    display: 'flex',
    cursor: 'pointer',
  },
  ({ right }) => ({
    [right ? 'marginRight' : 'marginLeft']: 15,
  }),
);

export const TTProgressBarArrow = glamorous.div<{
  right: boolean;
  disabled: boolean;
}>(
  {
    label: 'tt-progress_bar-arrow',
  },
  ({ right, disabled }) => ({
    border: `solid ${colours.$white}`,
    opacity: disabled ? 0.2 : 1,
    borderWidth: '0 3px 3px 0',
    display: 'inline-block',
    padding: 5,
    alignSelf: 'center',
    transform: right ? 'rotate(-45deg)' : 'rotate(135deg)',
  }),
);

export const TTVoteContainer = glamorous.div({
  label: 'tt-vote-container',
  height: '100%',
  width: '100%',
  display: 'flex',
  position: 'absolute',
  top: 0,
  backgroundColor: colours.$black,
  flexDirection: 'column',
  alignItems: 'center',
});

export const TTVoteHeadline = glamorous.div<{
  colour: string;
}>(
  {
    label: 'tt-vote-headline',
    fontFamily: fonts.$theSunMedium,
    textTransform: 'uppercase',
    marginTop: 10,
    fontSize: 18,
  },
  ({ colour }) => ({
    color: colour,
  }),
);

export const TTVoteSubHeadline = glamorous.div({
  label: 'tt-vote-sub_headline',
  fontFamily: fonts.$theSunHeavyNarrow,
  color: colours.$white,
  textTransform: 'uppercase',
  fontSize: '38px',
  flex: 2,
  width: '80%',
  textAlign: 'center',
  lineHeight: '40px',
  marginTop: 10,
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
  [mediaQueries.mobileInner]: {
    flex: 1,
  },
});

export const TTVoteBottomCopy = glamorous.div({
  label: 'tt-vote-bottom_copy',
  fontFamily: fonts.$theSunRegular,
  color: colours.$midGrey,
  fontSize: 16,
  width: '80%',
  flex: 1,
  textAlign: 'center',
  margin: '10px 0',
  display: 'flex',
  alignItems: 'center',
  justifyContent: 'center',
});

export const TTVoteButtonContainer = glamorous.div({
  label: 'tt-vote-button_container',
  display: 'flex',
  flexDirection: 'column',
  height: '60%',
  width: '100%',
  cursor: 'pointer',
  marginTop: '38%',
  [mediaQueries.mobileInner]: {
    marginTop: '18%',
  },
});

export const TTVoteButtonWrapper = glamorous.div({
  label: 'tt-vote-button-wrapper',
});

export const TTVoteButton = glamorous.a<{
  colour: string;
  hoverBackgroundColour: string;
}>(
  {
    label: 'tt-vote-button',
    display: 'flex',
    border: 0,
    padding: 0,
    height: '20%',
    marginTop: 10,
    cursor: 'pointer',
    textDecoration: 'none',
  },
  ({ colour, hoverBackgroundColour }) => ({
    backgroundColor: colour,
    ':hover': {
      backgroundColor: hoverBackgroundColour,
    },
  }),
);

export const TTVoteButtonThumbContainer = glamorous.div({
  label: 'tt-vote-button_thumb_container',
  flexBasis: '19%',
  [mediaQueries.mobileInner]: {
    flexBasis: '9%',
  },
});

export const TTVoteButtonThumb = glamorous.img({
  label: 'tt-vote-button_thumb',
  width: 'auto',
  height: '100%',
  display: 'block',
  border: `1px solid ${colours.$darkGrey}`,
});

export const TTVoteButtonText = glamorous.div<{ colour: string }>(
  {
    label: 'tt-vote-button_text',
    fontFamily: fonts.$theSunMedium,
    fontSize: 16,
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
    alignSelf: 'center',
    cursor: 'pointer',
  },
  ({ colour }) => ({
    color: colour,
  }),
);

export const TTResultsButtonLink = glamorous.div<{
  colour: string;
  hoverBackgroundColour: string;
}>(
  {
    label: 'tt-results-button_link',
    fontFamily: fonts.$theSunMedium,
    color: colours.$white,
    position: 'absolute',
    bottom: 15,
    fontSize: 16,
    height: 44,
    maxHeight: 44,
    justifyContent: 'center',
    alignItems: 'center',
    display: 'flex',
    flex: 1,
    width: '100%',
    cursor: 'pointer',
    [mediaQueries.mobileInner]: {
      width: 250,
    },
  },
  ({ colour, hoverBackgroundColour }) => ({
    backgroundColor: colour,
    ':hover': {
      backgroundColor: hoverBackgroundColour,
    },
  }),
);

export const TTResultsChartContainer = glamorous.div({
  label: 'tt-results-chart-container',
  width: '100%',
  height: '60%',
  marginTop: '39%',
  [mediaQueries.mobileInner]: {
    marginTop: '19%',
  },
});
