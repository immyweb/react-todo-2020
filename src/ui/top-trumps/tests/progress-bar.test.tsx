import * as React from 'react';
import { shallow } from 'enzyme';

import { ProgressBar } from '../progress-bar';
import { TTProgressBarCircle, TTProgressBarArrow, TTProgressBarArrowContainer } from '../styles';

const theme = {
  primary: '#ffffff',
  secondary: '#000000',
};

const cards = [
  {
    id: 'wilder',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg',
    fName: 'Deontay',
    sName: 'Wilder',
  },
  {
    id: 'whyte',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg',
    fName: 'Dillian',
    sName: 'Whyte',
  },
  {
    id: 'fury',
    imageMobile: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg',
    imageDesktop: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg',
    imageThumb: 'https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg',
    fName: 'Tyson',
    sName: 'Fury',
  },
];

function shallowRender(props?: any) {
  return shallow(
    <ProgressBar
      className={'youCantBuy'}
      cards={cards}
      theme={theme}
      currentCardIndex={0}
      onClickRight={() => {}}
      onClickLeft={() => {}}
      {...props}
    />,
  );
}

describe('<ProgressBar />', () => {
  it('renders correctly', () => {
    const wrapper = shallowRender();
    expect(wrapper).toMatchSnapshot();
  });

  it('renders correct number of items', () => {
    const wrapper = shallowRender();
    const items = wrapper.find(TTProgressBarCircle);
    expect(items.length).toEqual(cards.length);
  });

  it('disables previous arrow if on first slide', () => {
    const wrapper = shallowRender({ currentCardIndex: 0 });
    const prevArrow = wrapper.find(TTProgressBarArrow).first();
    expect(prevArrow.props().disabled).toEqual(true);
  });

  it('previous arrow is enabled if not on first slide', () => {
    const wrapper = shallowRender({ currentCardIndex: 1 });
    const prevArrow = wrapper.find(TTProgressBarArrow).first();
    expect(prevArrow.props().disabled).toEqual(false);
  });

  it('calls onClickLeft when left arrow is clicked and not disabled', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onClickLeft: fn, currentCardIndex: 1 });
    const btn = wrapper.find(TTProgressBarArrowContainer).first();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });

  it('does not call onClickLeft when left arrow is clicked and is disabled', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onClickLeft: fn, currentCardIndex: 0 });
    const btn = wrapper.find(TTProgressBarArrowContainer).first();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(0);
  });

  it('calls onClickRight when right arrow is clicked', () => {
    const fn = jest.fn();
    const wrapper = shallowRender({ onClickRight: fn });
    const btn = wrapper.find(TTProgressBarArrowContainer).last();
    btn.simulate('click');
    expect(fn).toHaveBeenCalledTimes(1);
  });
});
