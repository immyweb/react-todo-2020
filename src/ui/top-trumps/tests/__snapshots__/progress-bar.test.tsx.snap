// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProgressBar /> renders correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    cards={
      Array [
        Object {
          "fName": "Deontay",
          "id": "wilder",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WILDER@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/wilder-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-wilder@2x.jpg",
          "sName": "Wilder",
        },
        Object {
          "fName": "Dillian",
          "id": "whyte",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/WHYTE@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/whyte-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-whyte@2x.jpg",
          "sName": "Whyte",
        },
        Object {
          "fName": "Tyson",
          "id": "fury",
          "imageDesktop": "https://www.thesun.co.uk/wp-content/uploads/2018/08/FURY@2x.jpg",
          "imageMobile": "https://www.thesun.co.uk/wp-content/uploads/2018/08/fury-mob@2x.jpg",
          "imageThumb": "https://www.thesun.co.uk/wp-content/uploads/2018/08/thumbnail-fury@2x.jpg",
          "sName": "Fury",
        },
      ]
    }
    className="youCantBuy"
    currentCardIndex={0}
    onClickLeft={[Function]}
    onClickRight={[Function]}
    theme={
      Object {
        "primary": "#ffffff",
        "secondary": "#000000",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "bkgnd": "#000000",
      "borderColour": "#ffffff",
      "children": Array [
        <glamorous(div)
          onClick={[Function]}
          right={false}
        >
          <glamorous(div)
            disabled={true}
            right={false}
          />
        </glamorous(div)>,
        <glamorous(div)>
          <glamorous(div)
            className="tt-progress"
            inactiveColour="#ffffff"
            viewed={true}
          />
          <glamorous(div)
            className="tt-progress"
            inactiveColour="#ffffff"
            viewed={false}
          />
          <glamorous(div)
            className="tt-progress"
            inactiveColour="#ffffff"
            viewed={false}
          />
        </glamorous(div)>,
        <glamorous(div)
          onClick={[Function]}
          right={true}
        >
          <glamorous(div)
            disabled={false}
            right={true}
          />
        </glamorous(div)>,
      ],
      "className": "tt-progress_bar-container",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <glamorous(div)
            disabled={true}
            right={false}
          />,
          "onClick": [Function],
          "right": false,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": true,
            "right": false,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": Array [
            <glamorous(div)
              className="tt-progress"
              inactiveColour="#ffffff"
              viewed={true}
            />,
            <glamorous(div)
              className="tt-progress"
              inactiveColour="#ffffff"
              viewed={false}
            />,
            <glamorous(div)
              className="tt-progress"
              inactiveColour="#ffffff"
              viewed={false}
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": "wilder",
            "nodeType": "class",
            "props": Object {
              "className": "tt-progress",
              "inactiveColour": "#ffffff",
              "viewed": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "whyte",
            "nodeType": "class",
            "props": Object {
              "className": "tt-progress",
              "inactiveColour": "#ffffff",
              "viewed": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": "fury",
            "nodeType": "class",
            "props": Object {
              "className": "tt-progress",
              "inactiveColour": "#ffffff",
              "viewed": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "children": <glamorous(div)
            disabled={false}
            right={true}
          />,
          "onClick": [Function],
          "right": true,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "disabled": false,
            "right": true,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "bkgnd": "#000000",
        "borderColour": "#ffffff",
        "children": Array [
          <glamorous(div)
            onClick={[Function]}
            right={false}
          >
            <glamorous(div)
              disabled={true}
              right={false}
            />
          </glamorous(div)>,
          <glamorous(div)>
            <glamorous(div)
              className="tt-progress"
              inactiveColour="#ffffff"
              viewed={true}
            />
            <glamorous(div)
              className="tt-progress"
              inactiveColour="#ffffff"
              viewed={false}
            />
            <glamorous(div)
              className="tt-progress"
              inactiveColour="#ffffff"
              viewed={false}
            />
          </glamorous(div)>,
          <glamorous(div)
            onClick={[Function]}
            right={true}
          >
            <glamorous(div)
              disabled={false}
              right={true}
            />
          </glamorous(div)>,
        ],
        "className": "tt-progress_bar-container",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <glamorous(div)
              disabled={true}
              right={false}
            />,
            "onClick": [Function],
            "right": false,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": true,
              "right": false,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": Array [
              <glamorous(div)
                className="tt-progress"
                inactiveColour="#ffffff"
                viewed={true}
              />,
              <glamorous(div)
                className="tt-progress"
                inactiveColour="#ffffff"
                viewed={false}
              />,
              <glamorous(div)
                className="tt-progress"
                inactiveColour="#ffffff"
                viewed={false}
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": "wilder",
              "nodeType": "class",
              "props": Object {
                "className": "tt-progress",
                "inactiveColour": "#ffffff",
                "viewed": true,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "whyte",
              "nodeType": "class",
              "props": Object {
                "className": "tt-progress",
                "inactiveColour": "#ffffff",
                "viewed": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": "fury",
              "nodeType": "class",
              "props": Object {
                "className": "tt-progress",
                "inactiveColour": "#ffffff",
                "viewed": false,
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "class",
          "props": Object {
            "children": <glamorous(div)
              disabled={false}
              right={true}
            />,
            "onClick": [Function],
            "right": true,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "disabled": false,
              "right": true,
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
